api/Services/EmailService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Microsoft.Graph.Models;
+
+namespace Scv.Api.Services;
+
+public interface IEmailService
+{
+    Task<IEnumerable<Message>> GetFilteredEmailsAsync(string mailbox, string subjectPattern, string fromEmail);
+    Task<Dictionary<string, MemoryStream>> GetAttachmentsAsStreamsAsync(string mailbox, string messageId, string attachmentName);
+}
+
+public class EmailService(GraphServiceClient graphServiceClient) : IEmailService
+{
+    private readonly GraphServiceClient _graphServiceClient = graphServiceClient;
+
+    public async Task<IEnumerable<Message>> GetFilteredEmailsAsync(string mailbox, string subjectPattern, string fromEmail)
```

Comment:
Wondering whether this should be made more generic.  The fact it will only return emails with attachments is hidden inside the implementation.

To be more generic you should be able to filter based on any combination of:
- subjectPattern
- fromEmail
- containsAttachment

If any other the parameters are not specified, they are not included in the filtering.